src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,80 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int INCREASE_FACTOR = 2;
+
+    private Object[] keys;
+    private Object[] values;
+    private int currentCapacity;
+    private int currentSize;
+
+    public StorageImpl() {
+        keys = new Object[DEFAULT_CAPACITY];
+        values = new Object[DEFAULT_CAPACITY];
+        currentCapacity = DEFAULT_CAPACITY;
+        currentSize = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = getIndexOfKey(key);
+        if (index < 0) {
+            addNew(key, value);
+        } else {
+            values[index] = value;
+        }
     }
```

Comment:
Can we refactor this method?

Reply:
How? :)

Reply:
I think it is about the `else`  keyword, would be nice to structure our logic in a way to avoid `else`s by using the `return` or other things, because it usually brings complexity in readability
```suggestion
    @Override
    public void put(K key, V value) {
        int index = getIndexOfKey(key);
        if (index < 0) {
            addNew(key, value);
            return;
        }
        values[index] = value;
    }
```